File: WarehouseManagementWeb.Application/Dto/Input/ResourceShipment/CreateResourceShipmentInput.cs
namespace WarehouseManagementWeb.Application.Dto.Input.ResourceShipment
{
    /// <summary>
    /// Класс входной модели создания ресурса отгрузки.
    /// </summary>
    public class CreateResourceShipmentInput
    {
        /// <summary>
        /// Номер документа отгрузки.
        /// </summary>
        public string? DocumentShipmentNumberCode { get; set; }

        /// <summary>
        /// Дата документа отгрузки.
        /// </summary>
        public DateTime DocumentShipmentDate { get; set; } 

        /// <summary>
        ///  Id клиента документа отгрузки.
        /// </summary>
        public int DocumentShipmentClientId { get; set; }

        /// <summary>
        /// Список входящих ресурсов в документ отгрузки.
        /// </summary>
        public ICollection<IncludeResourceShipmentInput>? IncludeResourceShipmentInputs { get; set; }
    }
}

File: warehouse-management-web-angular/src/app/warehouse/document-shipment/models/input/create-resource-shipment-input.ts
import { IncludeResourceShipmentInput } from './include-resource-shipment-input';

/**
 * Класс входной модели создания ресурса отгрузки.
 */
export class CreateResourceShipmentInput {
  /**
   * Номер документа отгрузки.
   */
  documentShipmentNumberCode: string = '';
  /**
   * Дата документа отгрузки.
   */
  documentShipmentDate: Date = new Date();
  /**
   * Id клиента документа отгрузки.
   */
  documentShipmentClientId: number | null = null;

  /**
   * Нужно ли подписать документ отгрузки.
   */
  isSetActiveStatus: boolean = false;

  /**
   * Список входящих ресурсов в документ отгрузки.
   */
  includeResourceShipmentInputs: IncludeResourceShipmentInput[] = [];
}
